File: implementations/python/summation.py
import sys

SMALL  = 100
MEDIUM = 10000
LARGE  = 1000000

SIZES = {"small": SMALL, "medium": MEDIUM, "large": LARGE}

def summation(arr, n):
    total = 0
    for i in range(n):
        total += arr[i]
    return total

if __name__ == "__main__":
    n = SIZES[sys.argv[1]]
    arr = [i + 1 for i in range(n)]
    print(summation(arr, n))

File: implementations/javascript/summation.js
const SIZES = { small: 100, medium: 10000, large: 1000000 };
const n = SIZES[process.argv[2]];

function summation(arr, n) {
    let sum = 0;
    for (let i = 0; i < n; i++) sum += arr[i];
    return sum;
}

const arr = new Array(n);
for (let i = 0; i < n; i++) arr[i] = i + 1;
console.log(summation(arr, n));
